libparsec/crates/client/src/connection_monitor.rs

```diff
@@ -153,6 +106,72 @@ impl ConnectionMonitor {
     }
 }
 
+fn handle_sse_error(
+    state: &mut ConnectionState,
+    event_bus: &EventBus,
+    err: ConnectionError,
+) -> ControlFlow<()> {
+    set_closed_state(state, event_bus);
+
+    match err {
+        // The only legit error is if we couldn't reach the server...
+        ConnectionError::NoResponse(_) => ControlFlow::Continue(()),
+
+        // ...otherwise the server rejected us, hence there is no use
+        // retrying to connect and we just stop this coroutine
+        ConnectionError::ExpiredOrganization => {
+            event_bus.send(&EventExpiredOrganization);
+            ControlFlow::Break(())
+        }
+        ConnectionError::RevokedUser => {
+            event_bus.send(&EventRevokedUser);
+            ControlFlow::Break(())
+        }
+        ConnectionError::UnsupportedApiVersion {
+            api_version,
+            supported_api_versions,
+        } => {
+            let event = EventIncompatibleServer(IncompatibleServerReason::UnsupportedApiVersion {
+                api_version,
+                supported_api_versions,
+            });
+            event_bus.send(&event);
+            ControlFlow::Break(())
+        }
+        err => {
+            let event = EventIncompatibleServer(IncompatibleServerReason::Unexpected(err.into()));
+            event_bus.send(&event);
+            ControlFlow::Break(())
+        }
+    }
+}
+
+fn set_closed_state(state: &mut ConnectionState, event_bus: &EventBus) {
```

Comment:
`set_closed_state` and `set_opened_state` are pretty small and used only in a single place, so I'd rather have them inlined instead

Reply:
done